addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -224,3 +226,32 @@ def _update_l10n_es_tbai_submitted_xml(self, xml_doc, cancel):
 
     def _is_l10n_es_tbai_simplified(self):
         return self.commercial_partner_id == self.env.ref("l10n_es_edi_sii.partner_simplified")
+
+    def _get_vendor_bill_tax_values(self):
+        self.ensure_one()
+        results = defaultdict(lambda: {'base_amount': 0.0, 'tax_amount': 0.0})
+        amount_total = 0.0
+        for line in self.line_ids.filtered(lambda l: l.display_type in ('product', 'tax')):
+            if any(t.l10n_es_type == 'ignore' for t in line.tax_ids) or line.tax_line_id.l10n_es_type == 'ignore':
+                continue
+            if line.tax_line_id.l10n_es_type != 'retencion':
+                amount_total += line.balance
+            for tax in line.tax_ids.filtered(lambda t: t.l10n_es_type not in ('recargo', 'retencion')):
+                results[tax]['base_amount'] += line.balance
+
+            if ((tax := line.tax_line_id) and tax.l10n_es_type not in ('recargo', 'retencion') and
```

Comment:
Hello @jco-odoo 
Walrus operators are not allowed in 16.0, can you remove it asap? A test will prevent that in the future.

Reply:
@Xavier-Do I had seen that before already and your new runbot tests rightfully complained about it, but I think it has been fixed already.